inc/api.class.php

```diff
@@ -608,9 +608,7 @@ protected function getItem($itemtype, $id, $params = []) {
       }
 
       // retrieve computer softwares
-      if (isset($params['with_softwares'])
-          && $params['with_softwares']
-          && $itemtype == "Computer") {
+      if (isset($params['with_softwares']) && $params['with_softwares']) {
```

Comment:
As far as I've understood, `$itemtype` must be one of `$CFG_GLPI['software_types']`; this should be checked here